aws/resource_aws_appconfig_application.go

```diff
@@ -0,0 +1,177 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/appconfig"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsAppconfigApplication() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAppconfigApplicationCreate,
+		Read:   resourceAwsAppconfigApplicationRead,
+		Update: resourceAwsAppconfigApplicationUpdate,
+		Delete: resourceAwsAppconfigApplicationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 64),
+				),
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(0, 1024),
+				),
+			},
+			"tags": tagsSchema(),
+			"id": {
+				Type:     schema.TypeString,
```

Comment:
If we have to cast this later on, does it mean that it should be a different type?

Reply:
There's a lot of casting because AWS API calls take in pointers but terraform schema types are just the normal type. When we do casts like `aws.StringValue(app.Id)`, it just casts the `string` to `*string`. There are also a good number of casts going in the opposite direction.